event.go

```diff
@@ -138,12 +148,35 @@ func (eb *EventBuilder) Build(eventID string, now time.Time, origin ServerName,
 		PrevState *[]EventReference `json:"prev_state,omitempty"`
 	}
 	event.EventBuilder = *eb
-	if event.PrevEvents == nil {
-		event.PrevEvents = emptyEventReferenceList
+	eventFormat, err = roomVersion.EventFormat()
+	if err != nil {
+		return result, err
 	}
-	if event.AuthEvents == nil {
-		event.AuthEvents = emptyEventReferenceList
+	switch eventFormat {
+	case EventFormatV1:
+		if event.PrevEvents == nil {
+			event.PrevEvents = []EventReference{}
+		}
+		if event.AuthEvents == nil {
+			event.AuthEvents = []EventReference{}
+		}
+	case EventFormatV2:
+		resPrevEvents, resAuthEvents := []string{}, []string{}
+		switch prevEvents := event.PrevEvents.(type) {
+		case []EventReference:
+			for _, prevEvent := range prevEvents {
+				resPrevEvents = append(resPrevEvents, prevEvent.EventID)
+			}
+		}
+		switch authEvents := event.AuthEvents.(type) {
+		case []EventReference:
+			for _, authEvent := range authEvents {
+				resAuthEvents = append(resPrevEvents, authEvent.EventID)
```

Comment:
What? Why are we appending to `resPrevEvents`?

Reply:
Typo, fixed.